pkgs/development/tools/detekt/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchurl, makeWrapper, jre_headless }:
+stdenv.mkDerivation rec {
+  pname = "detekt";
+  version = "1.22.0";
+
+  src = fetchurl {
+    url = "https://github.com/detekt/detekt/releases/download/v${version}/detekt-cli-${version}-all.jar";
+    sha256 = "sha256-NCOMBcAtk7cOlP3H8Bz/hfR/305j/DfaoFrwc504b/4=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  phases = [ "installPhase" "checkPhase" ];
+
+  installPhase = ''
+    makeWrapper ${jre_headless}/bin/java $out/bin/${pname} --add-flags "-jar $src"
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+    $out/bin/${pname} --version | grep -q "${version}"
+  '';
```

Comment:
We have a helper for this: `testers.testVersion`

Reply:
👍 nice, I've switched to use that.